### PR TITLE
Bring spree_backend base controller to OFN 

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Related to #4050

Here we bring spree/admin/base_controller to OFN and merge its decorator.

One of the reasons we are doing this, we bring some code we need but end up deleting a good part of it or related useless stuff (like the alerts in this case) leaving us with basically the same number of lines of code:
Screenshot 2019-12-04 at 00 00 47

#### What should we test?
We remove a dead feature in spree (security alerts from spree itself). In general settings there was this checkbox, it's now gone:
Screenshot 2019-12-02 at 21 34 10

This controller is the parent of all controllers in spree.
This is what I think we should validate:
- navigation - menus and submenus work (we can test a few of them)
- changing language works in both frontoffice and backoffice
- authorization - try to access a URL a manager is not allowed with a manager logged in, like going to admin/general_settings/edit with a user who is just a manager
- api key - make sure that a page like orders list or products list basic functions work (this requires an api key to be in place)
- flash_message_for - make sure you see the flash message when successfully updating an enterprise
- banner - make sure that if you have an enterprise with an OC but without payment methods you see a flash message on top of the menu mentioned that problem

#### Release notes
Changelog Category: Changed
Moved basic admin controller code to OFN to remove dependency from Spree.

### CODE REVIEW

File: app/controllers/spree/admin/base_controller.rb [hunk 1]
```diff
@@ -19,9 +19,9 @@ class BaseController < Spree::BaseController
       def warn_invalid_order_cycles
         distributors = active_distributors_not_ready_for_checkout
 
-        if distributors.any? && flash[:notice].nil?
-          flash[:notice] = active_distributors_not_ready_for_checkout_message(distributors)
-        end
+        return unless distributors.any? && flash[:notice].nil?
```
[the remark this answers is not in the document]
Comment (luisramos0) [reply app/controllers/spree/admin/base_controller.rb]: yes, thanks :+1:

**Comment by luisramos0** [comment]
I removed a line of code I now realize is not needed. the helper method is never used in our code base.

**Comment by RachL** [comment]
>navigation - menus and submenus work (we can test a few of them)

All good.

>changing language works in both frontoffice and backoffice

All good.

>authorization - try to access a URL a manager is not allowed with a manager logged in, like going to admin/general_settings/edit with a user who is just a manager

All good 

image

>api key - make sure that a page like orders list or products list basic functions work (this requires an api key to be in place)

All good. Tested CRUD and bulk update.

>flash_message_for - make sure you see the flash message when successfully updating an enterprise 

All good.

image

>banner - make sure that if you have an enterprise with an OC but without payment methods you see a flash message on top of the menu mentioned that problem

@luisramos0 to my knowledge this banner only show up as super admin. I tried to create an OC without payment methods and now the new layout tell me I cannot select my hub if I don't have a payment method. I guess you added a test for this?

Anyway as this is impacting super admin only, I'm moving it to ready to go.

**Comment by luisramos0** [comment]
just to play on the safe side here, lets confirm we are on the same page.
Yes, the hubs cant be added to the OC without payment methods, that good.
BUT for example if you remove the last payment method from a hub already in an OC, you should see the banner as a manager...
If you dont see this banner in this case, it could be a bug, and it could have been introduced by this PR.

**Comment by RachL** [comment]
@luisramos0 yes but I can't delete a payment method once it is in the OC... Or you mean a fresh OC with no orders yet?

**Comment by luisramos0** [comment]
I didnt say delete, I said "remove the last payment method from a hub". I should have said "remove the hub from the list of enterprises associated with the payment method" :-) do you see what I mean?

**Comment by RachL** [comment]
Ok yes sorry I missunderstood. This is working:

image

Moving to ready to go!

**Comment by luisramos0** [comment]
ah, nice, thanks @RachL

**Pull request merged**